ci/date-check/src/main.rs

```diff
@@ -0,0 +1,221 @@
+use std::{
+    collections::BTreeMap,
+    convert::TryInto as _,
+    env, fmt, fs,
+    path::{Path, PathBuf},
+};
+
+use chrono::{Datelike as _, TimeZone as _, Utc};
+use glob::glob;
+use regex::Regex;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+struct Date {
+    year: u32,
+    month: u32,
```

Comment:
> I think we should limit this to `u8`

I tried that at first but it requires using `TryInto` because `chrono` uses `u32`.



> if not a custom enum for the specific twelve months

That seems like unnecessary complexity to me. At some point it might be good to have a `--validate` mode that makes sure all the dates are valid (e.g. not in the future) and put that in CI, but I don't think we should do that here.